migrations/versions/2025_01_21_0820-4dec3e456c9e_add_on_delete_cascade.py

```diff
@@ -0,0 +1,114 @@
+"""add_on_delete_cascade
+
+Revision ID: 4dec3e456c9e
+Revises: e6227073183d
+Create Date: 2025-01-21 08:20:12.221051+00:00
+
+"""
+
+from typing import Sequence, Union
+
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision: str = "4dec3e456c9e"
+down_revision: Union[str, None] = "e6227073183d"
+branch_labels: Union[str, Sequence[str], None] = None
+depends_on: Union[str, Sequence[str], None] = None
+
+
+def upgrade() -> None:
+    # To add ON DELETE CASCADE to the foreign key constraint, we need to
+    # rename the table, create a new table with the constraint, and copy
+    # the data over.
+    op.execute("ALTER TABLE prompts RENAME TO _prompts_old;")
+    op.execute(
+        """
+        CREATE TABLE prompts (
+            id TEXT PRIMARY KEY,  -- UUID stored as TEXT
+            timestamp DATETIME NOT NULL,
+            provider TEXT,       -- VARCHAR(255)
+            request TEXT NOT NULL,  -- Record the full request that arrived to the server
+            type TEXT NOT NULL, -- VARCHAR(50) (e.g. "fim", "chat")
+            workspace_id TEXT NOT NULL,
+            FOREIGN KEY (workspace_id) REFERENCES workspaces(id) ON DELETE CASCADE
+        );
+        """
+    )
+    op.execute("INSERT INTO prompts SELECT * FROM _prompts_old;")
+    op.execute("DROP TABLE _prompts_old;")
```

Comment:
Is all of this needed? Wouldn't you achieve something similar by dropping the foreign key constraint and re-creating it?